release/src/main/groovy/run_release_candidate_python_validation.sh

```diff
@@ -0,0 +1,272 @@
+if [[ "$hash_check" != "OK" ]]
+then
+  echo "ERROR: The md5 hash doesn't match."
+  complete "The md5 hash doesn't match."
+  exit 1
+fi
+echo "SUCCEED: Hashes verification completed."
+
+#
+# 3. create a new virtualenv and install the SDK
+#
+
+print_separator "Creating new virtualenv and installing the SDK"
+
+unzip $BEAM_PYTHON_RELEASE
+cd apache-beam-$VERSION/sdks/python/
+virtualenv temp_virtualenv
+. temp_virtualenv/bin/activate && python setup.py sdist && pip install dist/apache-beam-$VERSION.tar.gz[gcp]
+gcloud_version=$(gcloud --version | head -1 | awk '{print $4}')
+if [[ "$gcloud_version" < "189" ]]
+then
+  update_gcloud
+fi
+
+
+#
+# 4. Run wordcount with DirectRunner
+#
+
+print_separator "Running wordcount example with DirectRunner"
+python -m apache_beam.examples.wordcount --output wordcount_direct.txt
+file="wordcount_direct.txt-00000-of-00001"
+if [ -f "$file" ]
+then
+	echo "$file found."
+else
+	echo "ERROR: $file not found."
+	complete "failed when running wordcount example with DirectRunner."
+	exit 1
+fi
+echo "SUCCEED: wordcount successfully run on DirectRunner."
+
+
+#
+# 5. Run wordcount with DataflowRunner
+#
+
+print_separator "Running wordcount example with DataflowRunner "
+python -m apache_beam.examples.wordcount \
+    --output gs://$BUCKET_NAME/$WORDCOUNT_OUTPUT \
+    --staging_location gs://$BUCKET_NAME$TEMP_DIR \
+    --temp_location gs://$BUCKET_NAME$TEMP_DIR \
+    --runner DataflowRunner \
+    --job_name wordcount \
+    --project $PROJECT_ID \
+    --num_workers $NUM_WORKERS \
+    --sdk_location dist/apache-beam-$VERSION.tar.gz
+
+# verify results.
+gcs_pull_result=$(gsutil ls gs://$BUCKET_NAME)
+for index in {0..3}
+do
+    if [[ $gcs_pull_result != *"gs://$BUCKET_NAME/$WORDCOUNT_OUTPUT-0000$index-of-00004"* ]]
+    then
+        echo "ERROR: The wordcount example failed on DataflowRunner".
+        complete "failed when running wordcount example with DataflowRunner."
+        exit 1
+    fi
+done
+# clean output files from GCS
+gsutil rm gs://$BUCKET_NAME/$WORDCOUNT_OUTPUT-*
+echo "SUCCEED: wordcount successfully run on DataflowRunner."
+
+
+#
+# 6. Run Streaming wordcount with DirectRunner
+#
+
+create_pubsub
+print_separator "Running Streaming wordcount example with DirectRunner"
+python -m apache_beam.examples.streaming_wordcount \
+--input_topic projects/$STREAMING_PROJECT_ID/topics/$PUBSUB_TOPIC1 \
+--output_topic projects/$STREAMING_PROJECT_ID/topics/$PUBSUB_TOPIC2 \
+--streaming &
+pid=$!
+sleep 15
+
+# verify result
+run_pubsub_publish
+pull_result=$(run_pubsub_pull)
+should_see="Python: "
+if [[ $pull_result = *"$should_see"* ]]
+then
+    echo "SUCCEED: The streaming wordcount example running successfully on DirectRunner."
+else
+    echo "ERROR: The streaming wordcount example failed on DirectRunner."
+    cleanup_pubsub
+    kill -9 $pid
+    complete "failed when running streaming wordcount example with DirectRunner."
+    exit 1
+fi
+kill -9 $pid
+sleep 10
```

Comment:
why sleep here?

Reply:
I found the "kill -9 pid" doesn't kill the directrunner process immediately. Sometimes, it will terminate the process after the Dataflowrunner startup. I doubt this will impact the result of streaming on Dataflowrunner since they are sharing the same pubsub topics/subscription. Thus, sleep 10 sec here to make sure this process being killed in time.